catalogue/webapp/components/WorkDetails/WorkDetails.js

```diff
@@ -141,6 +141,29 @@ const WorkDetails = ({
   const locationOfWork = work.notes.find(
     note => note.noteType.id === 'location-of-original'
   );
+  const arrangementNote = work.notes.find(
+    note => note.noteType.id === 'arrangement-note'
+  );
+  const biographicalNote = work.notes.find(
+    note => note.noteType.id === 'biographical-note'
+  );
+  const relatedMaterial = work.notes.find(
+    note => note.noteType.id === 'related-material'
+  );
+  const acquisitionNote = work.notes.find(
+    note => note.noteType.id === 'acquisition-note'
+  );
+
+  const orderedNotes = [
+    arrangementNote,
+    acquisitionNote,
+    biographicalNote,
+    relatedMaterial,
+  ].filter(Boolean);
```

Comment:
There can be multiple notes for a single type, so I think this might deal with these cases incorrectly :thinking:

Reply:
Ah – thanks. I'll take a look.

Reply:
Could be something like this to avoid the multiple loops (I'm  sure could format the code slightly nicer but you get the idea):

```js
val notesOrdering = ["related-material", "arrangement-note", ...]
notes.sort((a, b) => { 
  let aIdx = notesOrdering.indexOf(a.type);
  let bIdx = notesOrdering.indexOf(b.type);
  if (aIdx == bIdx) return 0;
  if (aIdx == -1) return 1;
  return aIdx > bIdx ? 1 : -1 
})
```

Reply:
Do you think it's worth it? I feel like the readability might trump the optimisation. `¯\_(ツ)_/¯`

Reply:
Yeah I agree readability more important than premature optimisation. The nice thing about my method is that the `notesOrdering` variable makes things very declarative, and easy to add new ordered fields compared to your one, but the sort thing is obviously a mess. Your method definitely better on the whole for readability so maybe just go with that.